JCLOUDS-1108 workaround: GCE hardwareId short-form

aledsage opened the pull request.

duncangrant commented on locations/jclouds/src/main/java/org/apache/brooklyn/location/jclouds/JcloudsLocation.java:
```diff
@@ -1484,6 +1493,48 @@ public Template buildTemplate(ComputeService computeService, ConfigBag config, C
+        if (!"google-compute-engine".equals(provider)) {
+            return hardwareId;
+        }
+        if (hardwareId == null || hardwareId.toLowerCase().startsWith("http") || hardwareId.contains("/")) {
```
Should null check be separate?  It presumably doesn't represent a long-form id but rather a misuse of the function.

aledsage replied:
Agreed - given it's a private method, we can be stricter about what is passed in (e.g. throw `NullPointerException` on null) particularly as the one caller guards the call with `Strings.isNonBlank(config.get(HARDWARE_ID))`.

aledsage commented:
I've changed the null check, and squashed it into one commit. I've re-run all the (Live) tests in `JcloudsGceHardwareProfilesStubbedLiveTest` - they passed for me locally.

Once jenkins has run the unit tests again, I'll merge this.
